pkg/utils/overriding/overriding.go

```diff
@@ -116,6 +116,9 @@ func OverrideDevWorkspaceTemplateSpec(original *workspaces.DevWorkspaceTemplateS
 
 func ensureOnlyExistingElementsAreOverridden(spec *workspaces.DevWorkspaceTemplateSpecContent, overrides workspaces.Overrides) error {
 	return checkKeys(func(elementType string, keysSets []sets.String) []error {
+		if len(keysSets) <= 1 {
```

Comment:
Totally optional nit
```suggestion
		if len(keysSets) < 2 {
```